tawazi/node/node.py

```diff
@@ -156,6 +175,24 @@ def __repr__(self) -> str:
         """
         return f"{self.__class__.__name__} {self.id} ~ | <{hex(id(self))}>"
 
+    @property
+    def active(self) -> Union["UsageExecNode", bool]:
+        """Whether this ExecNode is active or not."""
+        # the value is set during the DAG description
+        if isinstance(self._active, UsageExecNode):
```

Comment:
I think it's weird to have an occurence of UsageExecNode in the father class, don't you think it would be better to override this property in UsageExecNode ? Also maybe we can make that an attribute in this case

Reply:
UsageExecNode is not a child of ExecNode

Reply:
Exactly. I think we should handle this afterwards in the PR fixing is_active and avoid the headaches of it for now